gap/prop.gi

```diff
@@ -8,7 +8,47 @@
 #############################################################################
 ##
 
-#
+InstallMethod(IsMeetSemilatticeDigraph, "for a digraph",
+[IsDigraph],
+function(digraph)
+  local out_nbrs, i, j, intr;
+  if IsPartialOrderDigraph(digraph) then
+    out_nbrs := OutNeighbours(digraph);
+    for i in [1 .. Size(out_nbrs)] do
+      for j in [i + 1 .. Size(out_nbrs)] do
+        # Is there a better way of doing this? JDM
+        intr := Intersection(out_nbrs[i], out_nbrs[j]);
+        if not intr in out_nbrs then
+          return false;
+        fi;
+      od;
+    od;
+    return true;
+  else
+    return false;
+  fi;
+end);
+
+InstallMethod(IsJoinSemilatticeDigraph, "for a digraph",
+[IsDigraph],
+function(digraph)
+  local in_nbrs, intr, i, j;
+  if IsPartialOrderDigraph(digraph) then
+    in_nbrs := InNeighbours(digraph);
+    for i in [1 .. Size(in_nbrs)] do
+      for j in [i + 1 .. Size(in_nbrs)] do
+        # Is there a better way of doing this? JDM
+        intr := Intersection(in_nbrs[i], in_nbrs[j]);
+        if not intr in in_nbrs then
+          return false;
+        fi;
+      od;
+    od;
```

Comment:
Yes I suppose so.

The way to solve this is to have two user-facing methods, `IsJoinSemilatticeDigraph` and `IsMeetSemilatticeDigraph`, and one hidden method `HiddenFunction` (say).

The hidden function implements what is currently implemented in `IsMeetSemilatticeDigraph`, except that it accepts a list of adjacencies rather than a digraph.

`IsMeetSemilatticeDigraph`  checks that the digraph is a partial order, then calls `HiddenFunction(OutNeighbours(digraph))`.

`IsJoinemilatticeDigraph`  checks that the digraph is a partial order, then calls `HiddenFunction(InNeighbours(digraph))`.

This uses no extra memory.

One of the first things that the method for `IsJoinSemilatticeDigraph` does is `in_nbrs := InNeighbours(digraph);`.
All that `DigraphReverse(digraph)` does is `Digraph(InNeighbours(digraph))`.